libs/ng-mocks/src/lib/resolve/collect-declarations.ts

```diff
@@ -316,6 +324,41 @@ const parseNgDef = (
   }
 };
 
+/**
+ * Note: This does not seem to work in every environment (e.g. the tests)
+ *       and is therefore a supplementary support for signals.
+ */
+const parseReflectComponentType = (def: any, declaration: Declaration): void => {
+  // Only available in NG 14+
+  const mirror = (angularCore as any)['reflectComponentType']?.(def);
+  if (!mirror) {
+    return;
+  }
+
+  if (mirror?.inputs) {
```

Comment:
The check for `if (mirror?.inputs)` is unnecessary here.  You already checked for mirror to be defined on line 334, and the `inputs` property of [ComponentMirror](https://angular.dev/api/core/ComponentMirror) should always be defined (albeit possibly an empty array).

Same goes for the check for `if (mirror?.outputs)` on line 350 below